### ISSUE TITLE
ABCL use of contrib/swank-repl.lisp fails

### ISSUE THREAD

**Issue by easye** [slime/slime#764]
@stassats recent work on SWANK-REPL contrib on Feb 15, 2023, viz. those commits through <https://github.com/slime/slime/commit/5e8fc7cad5ae7dc19b434c559ebaee3185d80b98> has borked ABCL.  

~~@dkochmanski 's work on <https://github.com/slime/slime/pull/760> has borked the use of the `contrib/slime-repl.lisp` with ABCL.~~  

I'll try to figure out how to get this working, but it involves papering over the incompleteness of ABCL's Gray streams implementation with errors of the form:

```
The value #<SWANK/GRAY::SLIME-INPUT-STREAM {4486D34}> is not of type STREAM.
```

**Comment by easye** [comment]
@stassats had accepted the previous non-conservative consolidation of the ASDF and `cl:load` paths for loading SLIME, but asked that inform/patch things when I had gotten a better Gray/ANSI/Java streams abstraction.  That work is still largely theoretical, unfortunately.

**Comment by stassats** [comment]
It would be a good idea to add ABCL to the CI.

**Comment by easye** [comment]
> It would be a good idea to add ABCL to the CI.

I'll look into it as part of verifying this fix.

**Comment by easye** [comment]
Some background on ABCL's streams <https://abcl.org/trac/wiki/DesignOfStreams>

**Comment by easye** [comment]
Whew: my patches to ABCL's Gray Streams implementation and our pretty printer (XP)  have matured to the point that I intend to release abcl-1.9.2 to solve this problem.   From this release on, ABCL should be a fully compliant provider of the implementations needed in the code in <file:swank/gray.lisp>.



TODO:

1.  Add ABCL to the CI to ensure this stays fixed.

2.  Add conditionals to warn that the <file:contrib/swank-repl.lisp> contrib will no longer work on versions lesser than abcl-1.9.2 with the next release of SLIME

-- 

Comments, suggestions on ways to proceed from @stassats always welcome.

**Comment by easye** [comment]
One thing that remains broken in my usage is that of the "external" SLIME contrib for telling emacs to render ANSI control sequences.  I would like to fix the remaining problems in ABCL in dealing with `cl:stream` IO which, I think, is a timing issue as ABCL tends to break up the ANSI control sequences in way that seems to confound Emacs.  Stay tuned Bear fans…

**Comment by easye** [comment]
> One thing that remains broken in my usage […]

I have determined that the problem here lies in the "external" `slime-repl-ansi-color` contrib, so this is no longer a blocker.

**Comment by easye** [comment]
As of <https://github.com/armedbear/abcl/commit/847fa63b615e5c9ae2c15d2f6392b91e96c634b9>, the GRAY-STREAMS support in ABCL has been improved to the point that `contrib/swank-repl.lisp` now works.  

I am in the process of releasing abcl-1.9.2 <<https://github.com/armedbear/abcl/issues/594>> which will contain this fix.

**Comment by zacque0** [comment]
> As of [armedbear/abcl@847fa63](https://github.com/armedbear/abcl/commit/847fa63b615e5c9ae2c15d2f6392b91e96c634b9), the GRAY-STREAMS support in ABCL has been improved to the point that `contrib/swank-repl.lisp` now works.
> 
> I am in the process of releasing abcl-1.9.2 <[armedbear/abcl#594](https://github.com/armedbear/abcl/issues/594)> which will contain this fix.

Thanks, can confirm that it works! Was facing the error just now. After compiling ABCL with the latest commit, it now works with SLIME.

My environment:
OS: Ubuntu 22.04.2 LTS x86_64
Java: 
```
openjdk 17.0.7 2023-04-18
OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7)
OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (build 17.0.7+7, mixed mode, sharing)
```
Emacs: GNU Emacs 28.2
Slime: slime-20230605.1004 (from MELPA)

**Comment by easye** [comment]
Closed with abcl-1.9.2
